packages/mui-material/src/TablePagination/TablePagination.js

```diff
@@ -248,7 +253,9 @@ const TablePagination = React.forwardRef(function TablePagination(inProps, ref)
           rowsPerPage={rowsPerPage}
           showFirstButton={showFirstButton}
           showLastButton={showLastButton}
+          slotProps={slotProps}
```

Comment:
Long term, we could add more things in the `slotProps`, for e.g. `toolbar`, `spacer` etc, so it would not be correct to just propagate the slotProps to the action component. In Base UI we treat the actions as a slot in the table pagination and provide the other slots there. I am not opposed to bringing these as first class slot keys in the TablePagination, however we also offer changing the component used for the Actions, via the `ActionComponent` prop (or what I would assume should be `slots.action`), so there may be different API in how these things are set. If we go with adding this new API, my proposal would to follow what we currently have in Base UI, or at least write down in an RFC what are the options, so that we can choose the best one. Happy to see other alternatives. cc @DiegoAndai

Reply:
> [...] my proposal would to follow what we currently have in Base UI

This makes sense to me. Here's the Base UI benchmark: https://mui.com/base-ui/react-table-pagination/#anatomy

So the change would be for the buttons props (e.g., `firstIconButton`) to be available:

- as `slotProps: { actions: slotProps: { firstIconButton: ... }` for the `TablePagination` component
- as { slotProps: { firstIconButton: ... }` for the `TablePaginationActions`

The reasoning behind the change is that the buttons are slots inside the actions slot, not direct TablePagination slots.

Is that correct?

Is this discussion enough, or should we create an RFC @mnajdova?

Another thing that came to mind. Should we remove the `Icon` from the name? e.g. `firstIconButton` -> `firstButton`, `previousIconButton` -> `previousButton`

Reply:
I went ahead and moved the `ActionsComponent` props into `slotProps.actions` in https://github.com/mui/material-ui/pull/39353/commits/b53c20a34c8ccee60bdecc98861a9696b237547f. Can make adjustments if needed.

Reply:
> Is that correct?

Yes

> Is this discussion enough, or should we create an RFC @mnajdova? I'm thinking an RFC is worth it to have more opinions.

Up to you, maybe we can phrase the RFC more around components that have high level components as slots, there are other similar use-cases.

> Another thing that came to mind. Should we remove the Icon from the name? e.g. firstIconButton -> firstButton, previousIconButton -> previousButton

Good call, we should also so it in Base UI, especially considering it does not bave the notion of icon buttons.

Reply:
> Up to you, maybe we can phrase the RFC more around components that have high level components as slots, there are other similar use-cases.

Let's move on with this change as it aligns with what Base UI already has. Later, we can have a higher-level discussion about nested slots.

> Good call, we should also so it in Base UI, especially considering it does not bave the notion of icon buttons.

Base [doesn't have `Icon` in its slots names](https://github.com/mui/material-ui/blob/master/packages/mui-base/src/TablePagination/TablePaginationActions.types.ts#L22), so not required 😊